api/draft_registrations/permissions.py

```diff
@@ -57,3 +59,27 @@ class DraftContributorDetailPermissions(ContributorDetailPermissions):
 
     def load_resource(self, context, view):
         return DraftRegistration.load(context['draft_id'])
+
+
+class DraftRegistrationPermission(permissions.BasePermission):
+    """
+    Check permissions for draft and node, Admin can create (POST) or edit (PATCH, PUT) to a DraftRegistration, but write
+    users can only edit them. Node permissions are inherited by the DraftRegistration when they are higher.
+    """
+    acceptable_models = (DraftRegistration, AbstractNode)
+
+    def has_object_permission(self, request, view, obj):
+        auth = get_user_auth(request)
+
+        if request.method in permissions.SAFE_METHODS:
+            if isinstance(obj, DraftRegistration):
+                return obj.is_contributor(auth.user) or obj.has_permission(auth.user, READ)
+            return obj.has_permission(auth.user, READ)
```

Comment:
I'd like this to be more specific. Maybe add an `isinstance(obj, NodeModel)` here, since we are checking the permission of the node?